### PR TITLE
generate helm values schema

### LINKED ISSUE #431: Add values schema to the k6-operator helm chart
### Feature Description

Hey, could we add the [values.schema.json](https://helm.sh/docs/topics/charts/#schema-files) to the helm chart? We are working on a [tool](https://github.com/cyclops-ui/cyclops) that renders UIs for helm charts based on their schema and would love to support the k6-operator chart as well.

### Suggested Solution (optional)

Add the values.schema.json to the chart. Initially, the schema can cover just the basic set of fields, so it is easier to maintain.

### Already existing or connected issues / PRs (optional)

_No response_

### PR DESCRIPTION
**Pull request by petar-cvit**
closes #431

Adds initial `schema.values.json` to the chart and extends `helm-release` with a step that generates and commits the schema.

Also, adds a new Makefile target to generate the schema using [github.com/dadav/helm-schema](github.com/dadav/helm-schema).

### CODE REVIEW

**Review by yorugac — CHANGES_REQUESTED** [review]
Hi @petar-cvit, thank you for the PR!

The schema makes sense and looks reasonable, IMHO, except for one field marked as required: please see my comment.

The CI probably won't work out-of-the-box :sweat_smile: Esp. tied to the release. I'll need to test it from the main branch and probably tinker with it anyway, but thanks for the base to build on :raised_hands:

**Comment by yorugac** [comment]
Also, @petar-cvit, please sign the CLA!

File: charts/k6-operator/values.schema.json [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+              "default": "",
+              "description": "global.image.registry -- Global image registry to use if it needs to be overridden for some specific use cases (e.g local registries, custom images, ...)",
+              "title": "registry",
+              "type": "string"
+            }
+          },
+          "title": "image",
+          "type": "object",
+          "required": [
+            "pullSecrets"
```
[the remark this answers is not in the document]
Comment (petar-cvit) [reply charts/k6-operator/values.schema.json]: All fields are required by default. I added the `required: false` annotation to mark it as not required

**Comment by yorugac** [comment]
@petar-cvit, thanks for the updates! I don't have any other change requests except: CLA is marked as not signed?
image

**Comment by petar-cvit** [comment]
@yorugac, sorry for the delay. CLA signed

**Review by yorugac — APPROVED** [review]
LGTM! Thank you @petar-cvit :slightly_smiling_face:

**Pull request merged**